src/data_provider/src/packages/appxWindowsWrapper.h

```diff
@@ -0,0 +1,401 @@
+         *
+         * @return Returns the vendor name
+         */
+        const std::string getVendor(const Utils::Registry& registry)
+        {
+            std::string vendor;
+
+            for (const auto& folder : registry.enumerate())
+            {
+                try
+                {
+                    const Utils::Registry fileReg(m_key, m_userId + "\\" + APPLICATION_STORE_REGISTRY + "\\" + m_nameApp + "\\" + folder + FILE_ASSOCIATIONS_REGISTRY, KEY_READ | KEY_QUERY_VALUE);
+                    vendor = searchPublisher(fileReg);
+
+                    if (vendor.empty())
+                    {
+                        const Utils::Registry urlReg(m_key, m_userId + "\\"  + APPLICATION_STORE_REGISTRY + "\\" + m_nameApp + "\\" + folder + URL_ASSOCIATIONS_REGISTRY, KEY_READ | KEY_QUERY_VALUE);
+                        vendor = searchPublisher(urlReg);
+                    }
+                }
+                catch (...)
+                {
+                }
+            }
+
+            return vendor;
+        }
+
+        /*
+         * @brief Search the name application in the cache registry.
+         *
+         * @param[in] nameApp Abbreviation of the name application to look for in the cache registry.
+         * @param[in] nameKey Name key where is the name application
+         *
+         * @return Return the name application.
+         */
+        const std::string searchNameFromCacheRegistry(const std::string& nameApp, const std::string& nameKey)
+        {
+            std::string registry;
+            std::string name;
+            constexpr auto INVALID_NAME_APP { "@{" };
+            auto findCacheName
+            {
+                [&nameApp](const std::string folder)
+                {
+                    return folder.find(nameApp) != std::string::npos;
+                }
+            };
+
+            const auto folder = std::find_if(m_cacheReg.begin(), m_cacheReg.end(), findCacheName);
+            // Search for the name of the folder containing the application name
+
+            if (folder != m_cacheReg.end())
+            {
+                registry = *folder;
+            }
+
+            if (!registry.empty())
+            {
+                name = searchKeyOnSubRegistries(m_userId + "\\" + CACHE_NAME_REGISTRY + "\\" + registry, nameKey);
+            }
+
+            const auto index { name.find(INVALID_NAME_APP) };
+            return (index != std::string::npos && index == 0) ? "" : name;
+        }
+
+        /*
+         * @brief Search the key on the registry and sub-registries.
+         *
+         * @param[in] path Start path where you want to search.
+         * @param[in] key Name key to look for
+         *
+         * @return Return the value of the key.
+         */
+        const std::string searchKeyOnSubRegistries(const std::string& path, const std::string& key)
+        {
+            std::string value;
+            const Utils::Registry registry(m_key, path);
```

Comment:
registry.enumerate() can throw an exception let's catch it please and return the default value.